packages/main/src/Menu.ts

```diff
@@ -341,27 +345,46 @@ class Menu extends UI5Element {
 	}
 
 	_itemKeyDown(e: KeyboardEvent) {
-		if (!isLeft(e) && !isRight(e)) {
-			return;
-		}
-
-		const shouldCloseMenu = this.isRtl ? isRight(e) : isLeft(e);
-		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
+		const isTabNextPrevoius = isTabNext(e) || isTabPrevious(e);
 		const item = e.target as MenuItem;
 		const parentElement = item.parentElement as MenuItem;
+		const shouldCloseMenu = parentElement.hasAttribute("ui5-menu-item") && (this.isRtl ? isRight(e) : isLeft(e));
+		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
 
-		if (isEnter(e)) {
-			e.preventDefault();
-		}
-		if (shouldOpenMenu) {
-			this._openItemSubMenu(item);
-		} else if (shouldCloseMenu && parentElement.hasAttribute("ui5-menu-item") && parentElement._popover) {
-			parentElement._popover.open = false;
-			parentElement.selected = false;
-			(parentElement._popover.opener as HTMLElement)?.focus();
+		if (item.hasAttribute("ui5-menu-item")) {
+			if (isEnter(e) || isTabNextPrevoius) {
+				e.preventDefault();
+			}
+
+			if (isRight(e)) {
+				item._navigateToEndContent();
+			} else if (isLeft(e)) {
+				item._navigateToEndContent(true);
+			}
+
+			if (shouldOpenMenu) {
+				this._openItemSubMenu(item);
+			} else if ((shouldCloseMenu || isTabNextPrevoius) && parentElement._popover) {
+				parentElement._popover.open = false;
```

Comment:
this in future should also be changed. if we want to change "parent" property, we fire an event and the parent changes the property based on that.

Reply:
Agree, I didn't mention it because it was from the "old code" but this whole thing must be changed some time in the future.